src/box/iproto.cc

```diff
@@ -2176,29 +2158,138 @@ iproto_session_push(struct session *session, struct port *port)
+{
+	iproto_thread->destroy_route[0] =
+		{ tx_process_destroy, &iproto_thread->net_pipe };
+	iproto_thread->destroy_route[1] =
+		{ net_finish_destroy, NULL };
+	iproto_thread->disconnect_route[0] =
+		{ tx_process_disconnect, &iproto_thread->net_pipe };
+	iproto_thread->disconnect_route[1] =
+		{ net_finish_disconnect, NULL };
+	iproto_thread->misc_route[0] =
+		{ tx_process_misc, &iproto_thread->net_pipe };
+	iproto_thread->misc_route[1] = { net_send_msg, NULL };
+	iproto_thread->call_route[0] =
+		{ tx_process_call, &iproto_thread->net_pipe };
+	iproto_thread->call_route[1] = { net_send_msg, NULL };
+	iproto_thread->select_route[0] =
+		{ tx_process_select, &iproto_thread->net_pipe };
+	iproto_thread->select_route[1] = { net_send_msg, NULL };
+	iproto_thread->process1_route[0] =
+		{ tx_process1, &iproto_thread->net_pipe };
+	iproto_thread->process1_route[1] = { net_send_msg, NULL };
+	iproto_thread->sql_route[0] =
+		{ tx_process_sql, &iproto_thread->net_pipe };
+	iproto_thread->sql_route[1] = { net_send_msg, NULL };
+	iproto_thread->join_route[0] =
+		{ tx_process_replication, &iproto_thread->net_pipe };
+	iproto_thread->join_route[1] = { net_end_join, NULL };
+	iproto_thread->subscribe_route[0] =
+		{ tx_process_replication, &iproto_thread->net_pipe };
+	iproto_thread->subscribe_route[1] = { net_end_subscribe, NULL };
+	iproto_thread->error_route[0] =
+		{ tx_reply_iproto_error, &iproto_thread->net_pipe };
+	iproto_thread->error_route[1] = { net_send_error, NULL };
+	iproto_thread->push_route[0] =
+		{ iproto_process_push, &iproto_thread->tx_pipe };
+	iproto_thread->push_route[1] = { tx_end_push, NULL };
+	/* IPROTO_OK */
+	iproto_thread->dml_route[0] = NULL;
+	/* IPROTO_SELECT */
+	iproto_thread->dml_route[1] = iproto_thread->select_route;
+	/* IPROTO_INSERT */
+	iproto_thread->dml_route[2] = iproto_thread->process1_route;
+	/* IPROTO_REPLACE */
+	iproto_thread->dml_route[3] = iproto_thread->process1_route;
+	/* IPROTO_UPDATE */
+	iproto_thread->dml_route[4] = iproto_thread->process1_route;
+	/* IPROTO_DELETE */
+	iproto_thread->dml_route[5] = iproto_thread->process1_route;
+	 /* IPROTO_CALL_16 */
+	iproto_thread->dml_route[6] =  iproto_thread->call_route;
+	/* IPROTO_AUTH */
+	iproto_thread->dml_route[7] = iproto_thread->misc_route;
+	/* IPROTO_EVAL */
+	iproto_thread->dml_route[8] = iproto_thread->call_route;
+	/* IPROTO_UPSERT */
+	iproto_thread->dml_route[9] = iproto_thread->process1_route;
+	/* IPROTO_CALL */
+	iproto_thread->dml_route[10] = iproto_thread->call_route;
+	/* IPROTO_EXECUTE */
+	iproto_thread->dml_route[11] = iproto_thread->sql_route;
+	/* IPROTO_NOP */
+	iproto_thread->dml_route[12] = NULL;
+	/* IPROTO_PREPARE */
+	iproto_thread->dml_route[13] = iproto_thread->sql_route;
+	iproto_thread->connect_route[0] =
+		{ tx_process_connect, &iproto_thread->net_pipe };
+	iproto_thread->connect_route[1] = { net_send_greeting, NULL };
+
+	slab_cache_create(&iproto_thread->net_slabc, &runtime);
```

Comment:
It looks a bit weird that in function with name _init_routes not only routes are intialized: I mean also slab cache, rmean etc. So either name it simply _init or split it into two funcs.

Reply:
Fixed